user-operator/src/test/java/io/strimzi/operator/user/UserOperatorConfigTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class UserOperatorConfigTest {
+    private static Map<String, String> envVars = new HashMap<>(5);
+    private static Map<String, String> expectedLabels = new HashMap<>(2);
+
+    static {
+        envVars.put(UserOperatorConfig.STRIMZI_NAMESPACE, "namespace");
+        envVars.put(UserOperatorConfig.STRIMZI_FULL_RECONCILIATION_INTERVAL_MS, "30000");
+        envVars.put(UserOperatorConfig.STRIMZI_LABELS, "label1=value1,label2=value2");
+        envVars.put(UserOperatorConfig.STRIMZI_CA_NAME, "ca-secret");
+        envVars.put(UserOperatorConfig.STRIMZI_CA_NAMESPACE, "differentnamespace");
+
+        expectedLabels.put("label1", "value1");
+        expectedLabels.put("label2", "value2");
+    }
+
+    @Test
+    public void testConfig()    {
+        UserOperatorConfig config = UserOperatorConfig.fromMap(envVars);
+
+        assertEquals(envVars.get(UserOperatorConfig.STRIMZI_NAMESPACE), config.getNamespace());
+        assertEquals(Long.parseLong(envVars.get(UserOperatorConfig.STRIMZI_FULL_RECONCILIATION_INTERVAL_MS)), config.getReconciliationIntervalMs());
+        assertEquals(expectedLabels, config.getLabels());
+        assertEquals(envVars.get(UserOperatorConfig.STRIMZI_CA_NAME), config.getCaName());
+        assertEquals(envVars.get(UserOperatorConfig.STRIMZI_CA_NAMESPACE), config.getCaNamespace());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
I wonder whether we should be using IAE here. Really the error occurs because the env var is not set (rather than it having an illegal value). Maybe we should have a common `InvalidConfigurationException` which all the operators throw when their environment isn't configured correctly.